services/storage/virtualharddisk/wssd.go

```diff
@@ -116,7 +117,7 @@ func getVirtualHardDiskPrecheckRequest(group, container string, vhds []*storage.
 	return request, nil
 }
 
-func getVirtualHardDiskRequest(opType wssdcloudcommon.Operation, group, container, name string, storage *storage.VirtualHardDisk) (*wssdcloudstorage.VirtualHardDiskRequest, error) {
+func getVirtualHardDiskRequest(opType wssdcloudcommon.Operation, group, container, name string, storage *storage.VirtualHardDisk, sourcePath string, sourceType common.ImageSource) (*wssdcloudstorage.VirtualHardDiskRequest, error) {
```

Comment:
I wonder if you need to update the function signature for precheck too for this type of create.

Reply:
Have you talked to @JamesLiang29 he was implementing precheck foir VHD. https://github.com/microsoft/wssdcloudagent/pull/1029/files

Reply:
Reached out to him. As his cloudagent PR is not checked in yet and feature is not live will work with him to enable precheck for this scenario too.